### PR TITLE
Allow privateers to work in mining rooms

### PR DESCRIPTION
**Pull request by Dasku**
Allows privateers to use containers instead of harvesting when sent to a room being remote mined.

### CODE REVIEW

**Comment by cyberblast** [comment]
but that is already included in robbing...

**Comment by Dasku** [comment]
hmm sorry.

**Comment by whardeman** [comment]
@cyberblast Would that not require the robbing flag to be set as well?

**Comment by cyberblast** [comment]
oh, yes you are right. it's currently validating on that in isValidAction of the robbing action. 
We could maybe remove that validation...

File: creep.action.robbing.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var action = new Creep.Action('robbing');
 action.maxPerTarget = 2;
 action.maxPerAction = 10;
 action.isValidAction = function(creep){
-    return ( creep.sum < creep.carryCapacity && (FlagDir.find(FLAG_COLOR.invade.robbing, creep.pos, true) != null) );
+    return ( creep.sum < creep.carryCapacity );
```
Comment (cyberblast) [inline creep.action.robbing.js]: or we could do `return ( creep.sum < creep.carryCapacity && !creep.room.my);` to allow robbing in any non-owned room, preventing privateers to steal from our homes...

**Pull request merged**